### DISCUSSION TITLE
Need Clarity: why need gspt.SetProcTitle(os.Args[0] + " server") when server run

### DISCUSSION THREAD

**Discussion by xiaods** [k3s-io/k3s#4501] [Q&A] [answered]
I am reading the source code for k3s, I don't know why need this line 
gspt.SetProcTitle(os.Args[0] + " server") 

anyone know it and give some hints.

**Comment by xiaods** [comment]
I'm try build cross compile based on AMD64 image. but it block by this line:
https://github.com/k3s-io/k3s/blob/master/scripts/build#L106
```bash
CGO_ENABLED=1 "${GO}" build -tags "$TAGS" -ldflags "$VERSIONFLAGS $LDFLAGS $STATIC_SQLITE" -o bin/containerd ./cmd/server/main.go
```
CGO_ENABLED=1 will block cross compile. so I search and found only gspt.SetProcTitle use C func.

**Comment by xiaods** [comment]
CGO_ENABLED=1 go build -trimpath   -tags "apparmor seccomp selinux netgo osusergo" -ldflags "-extldflags -static -X main.gitCommit=v1.0.2-0-g52b36a2d -X main.version=1.0.2 -w -s" -o runc .

**Comment by brandond** [comment, marked as answer]
K3s must be built with CGO_ENABLED; doing otherwise is not supported.
https://github.com/k3s-io/k3s/blob/535a9196351b65764071a27b7d68ec7497e67e55/scripts/build#L105-L106

The reason for calling setproctitle is discussed at https://github.com/k3s-io/k3s/issues/2014
